tests/ludwig/encoders/test_text_encoders.py

```diff
@@ -1,55 +1,133 @@
+    from transformers import AutoConfig, AutoModel
+
+    config = AutoConfig.from_pretrained(pretrained_model_name_or_path)
+    return AutoModel.from_config(config)
+
+
+@pytest.fixture
+def mock_load_encoder_from_hf_hub(monkeypatch):
+    """Mocks encoder downloads from HuggingFace Hub.
+
+    With this mock, only encoder configs are downloaded, not the encoder weights.
+    """
+    monkeypatch.setattr(text_encoders, "load_pretrained_hf_model", _load_pretrained_hf_model_no_weights)
+
+
+def get_mismatched_config_params(ludwig_results_dir, ludwig_model):
+    saved_config_dict = load_json(os.path.join(ludwig_results_dir, "model", MODEL_HYPERPARAMETERS_FILE_NAME))
+    saved_config_obj = ModelConfig.from_dict(saved_config_dict)
+
+    mismatches = []
+    for input_feature_config in saved_config_obj.input_features.to_list():
+        feature_name = input_feature_config[NAME]
+        encoder_config_from_file = input_feature_config[ENCODER]
+        encoder_config_from_model = ludwig_model.model.input_features[feature_name].encoder_obj.config.to_dict()
+        for k, v in encoder_config_from_model.items():
+            # Skip saved_weights_in_checkpoint because this value is not yet set when the global config
+            # is modified with the final encoder config.
+            if k == "saved_weights_in_checkpoint":
+                continue
+
+            if encoder_config_from_file[k] != v:
+                mismatch = {
+                    "feature_name": feature_name,
+                    "param_name": k,
+                    "val_from_file": encoder_config_from_file[k],
+                    "val_from_model": v,
+                }
+                mismatches.append(mismatch)
+    return mismatches
 
 
 @pytest.mark.parametrize(
     "encoder_config_cls",
     [
+        configs.AutoTransformerConfig,
         configs.ALBERTConfig,
         configs.BERTConfig,
         configs.XLMConfig,
-        configs.GPTConfig,
+        pytest.param(configs.GPTConfig, marks=pytest.mark.skip("Causes exit code 143 in CI")),
         configs.RoBERTaConfig,
         configs.GPT2Config,
         configs.DistilBERTConfig,
         configs.TransformerXLConfig,
-        configs.CTRLConfig,
+        pytest.param(configs.CTRLConfig, marks=pytest.mark.skip("Disabled in the schema")),
         configs.CamemBERTConfig,
-        configs.MT5Config,
+        pytest.param(configs.MT5Config, marks=pytest.mark.skip("Disabled in the schema")),
         configs.XLMRoBERTaConfig,
         configs.LongformerConfig,
         configs.ELECTRAConfig,
         configs.FlauBERTConfig,
         configs.T5Config,
         configs.XLNetConfig,
-        configs.DistilBERTConfig,
     ],
 )
-def test_hf_pretrained_default_exists(encoder_config_cls: configs.SequenceEncoderConfig):
-    """Test that the default pretrained model exists on the HuggingFace Hub.
+def test_hf_ludwig_model_e2e(tmpdir, csv_filename, mock_load_encoder_from_hf_hub, encoder_config_cls):
```

Comment:
With this test, do we still need the these encoder class-level tests: https://github.com/ludwig-ai/ludwig/blob/master/tests/ludwig/encoders/test_text_encoders.py#L82-L308

My understanding is that combiner class instantiation relies on input feature output shape, and that a full forward pass would require the input feature output shape to match with the combiner's dimensions. Since this is an end-to-end test that tests model training, it seems like we would achieve equal coverage with the end-to-end test alone.

If we can remove the other tests, it would greatly reduce the CI run time. Happy to leave it if people think there's a reason to though.

Reply:
Let's remove the other tests since the text encoders are covered by this test, and reducing the CI runtime would be a big win!

Reply:
Sounds good! I also found these tests: https://github.com/ludwig-ai/ludwig/blob/master/tests/integration_tests/test_experiment.py#L88-L96

I'm going to update these new tests to incorporate the functionality in those tests and delete them as well (i.e. using `LudwigModel.experiment` instead of `LudwigModel.train`). I believe the tests are semantically equivalent.